python_modules/dagster/dagster/_core/definitions/cacheable_assets.py

```diff
@@ -0,0 +1,78 @@
+from typing import AbstractSet, Dict, FrozenSet, Mapping, NamedTuple, Optional, Sequence
+
+import dagster._check as check
+from dagster._core.definitions.assets import AssetsDefinition
+from dagster._core.definitions.events import AssetKey
+from dagster._core.definitions.metadata import MetadataUserInput
+from dagster._serdes import whitelist_for_serdes
+from dagster._utils import frozendict
+
+
+@whitelist_for_serdes
+class AssetsDefinitionMetadata(
+    NamedTuple(
+        "_PipelineSnapshot",
+        [
+            ("input_keys", FrozenSet[AssetKey]),
+            ("output_keys", FrozenSet[AssetKey]),
+            ("asset_deps", Optional[Mapping[AssetKey, FrozenSet[AssetKey]]]),
```

Comment:
as below, `AssetKey`-ed map doesn't seem to play nice with GRPC JSON marshalling